packages/core/src/lib/wallet/wallet.types.ts

```diff
@@ -121,6 +128,112 @@ interface SignAndSendTransactionsParams {
   transactions: Array<Optional<Transaction, "signerId">>;
 }
 
+interface SignTransactionParams {
+  /**
+   * The Transaction object to sign
+   */
+  transaction: Tx;
+  /**
+   * The {Signer} object that assists with signing keys
+   */
+  signer: Signer;
+  /**
+   * The human-readable NEAR account name
+   */
+  accountId?: string;
+  /**
+   * The targeted network. (ex. default, betanet, etc…)
+   */
+  networkId?: string;
+}
+
+interface SignTransactionActionsParams {
+  /**
+   * The NEAR account ID of the transaction receiver.
+   */
+  receiverId: string;
+  /**
+   * Tx nonce.
+   */
+  nonce: bigint;
```

Comment:
Tx nonce is attached to the public key nonce of the account, it does not make sense to require it from the app developer.